index.js

```diff
@@ -216,197 +170,102 @@ class XiaomiRoborockVacuum {
       }
     }
 
-    // Declare services for rooms in advance, so HomeKit can create the switches
-    if (this.config.autoroom && Array.isArray(this.config.autoroom)) {
-      for (const i in this.config.autoroom) {
-        // Index will be overwritten, when robot is available
-        this.createRoom(i, this.config.autoroom[i]);
-      }
-    }
-
     if (this.config.zones) {
       for (var i in this.config.zones) {
         this.createZone(this.config.zones[i].name, this.config.zones[i].zone);
       }
     }
+    //Changed by Mai
+    if (this.config.gotoTarget) {
+      if (this.config.gotoTarget.target && this.config.gotoTarget.target.length == 2) {
+        var name = this.config.gotoTarget.name || 'Go to';
+        this.services.gotoTarget = new Service.Switch(name, 'GotoTargetSwitch');
+        this.services.gotoTarget.getCharacteristic(Characteristic.On).on('get', (cb) => callbackify(() => this.getGotoTarget(), cb)).
+          on('set', (newState, cb) => callbackify(() => this.setGotoTarget(newState, this.config.gotoTarget.target), cb));
+      }
+    }
```

Comment:
Actual change to include in a PR

Reply:
The goToTarget function, this change is isolated from the rest so can be safely merged.